packages/components/src/form-token-field/index.js

```diff
@@ -48,13 +48,20 @@ class FormTokenField extends Component {
 		this.onInputChange = this.onInputChange.bind( this );
 		this.bindInput = this.bindInput.bind( this );
 		this.bindTokensAndInput = this.bindTokensAndInput.bind( this );
+		this.updateSuggestions = this.updateSuggestions.bind( this );
 	}
 
-	componentDidUpdate() {
+	componentDidUpdate( prevProps ) {
 		// Make sure to focus the input when the isActive state is true.
 		if ( this.state.isActive && ! this.input.hasFocus() ) {
 			this.input.focus();
 		}
+
+		const { suggestions, value } = this.props;
+		const suggestionsDidUpdate = suggestions !== prevProps.suggestions;
```

Comment:
This probably updates more than you're expecting it to, at least based on the shape of `suggestions` being an array, and the array being generated as a new value on each render.

Essentially, it comes down to:

```js
const a = [ 'foo' ];
const b = [ 'foo' ];
console.log( a === b );
// false
```

Related: http://adripofjavascript.com/blog/drips/object-equality-in-javascript.html

At least in how this is used by `FlatTermSelector`, the array is a result of `Array#map`, so would be a new value on each render:

https://github.com/WordPress/gutenberg/blob/8960509d97f72bf4bee68e1513a6c3fd3b0d446e/packages/editor/src/components/post-taxonomies/flat-term-selector.js#L211

https://github.com/WordPress/gutenberg/blob/8960509d97f72bf4bee68e1513a6c3fd3b0d446e/packages/editor/src/components/post-taxonomies/flat-term-selector.js#L229

>The map() method **creates a new array** with the results of calling a provided function on every element in the calling array.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Options:

- Accept the fact that it renders more than we expect it to, but that it might benefit from _some_ reuse if the references stay the same
- Abandon any attempt to avoid updating and always update suggestions
- Consider instead equality of members for sameness ([`@wordpress/is-shallow-equal`](https://github.com/WordPress/gutenberg/tree/master/packages/is-shallow-equal)). There is a small performance overhead to this, if it's worth considering the costliness vs. `updateSuggestions` (or whether `updateSuggestions` must only be called when in-fact it changes)

Reply:
Hey @aduth - thanks for the feedback, and sorry for the super late reply. I missed this comment (and only read the review one further down). 🤦‍♂

You are absolutely right, and I now implemented `isShallowEqual` to compare suggestions. 👍

Is this PR all good now? (Build and tests are still pasing.) 🙂